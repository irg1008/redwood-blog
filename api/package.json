{
  "name": "api",
  "version": "0.0.0",
  "private": true,
  "dependencies": {
    "@paralleldrive/cuid2": "2.2.2",
    "@react-email/components": "0.0.31",
    "@redwoodjs/api": "8.4.1",
    "@redwoodjs/api-server": "8.4.1",
    "@redwoodjs/auth-dbauth-api": "8.4.1",
    "@redwoodjs/graphql-server": "8.4.1",
    "@redwoodjs/jobs": "8.4.1",
    "@redwoodjs/mailer-core": "8.4.1",
    "@redwoodjs/mailer-handler-nodemailer": "8.4.1",
    "@redwoodjs/mailer-renderer-react-email": "8.4.1",
    "@redwoodjs/realtime": "8.4.1",
    "cookie": "1.0.2",
    "i18next": "24.1.0",
    "i18next-http-middleware": "3.7.0",
    "ioredis": "5.4.1",
    "jose": "5.9.6",
    "schemas": "workspace:*"
  },
  "devDependencies": {
    "@redwoodjs/mailer-handler-in-memory": "8.4.1",
    "@redwoodjs/mailer-handler-studio": "8.4.1"
  },
  "eslintConfig": {
    "rules": {
      "react/react-in-jsx-scope": "off"
    }
  }
}
