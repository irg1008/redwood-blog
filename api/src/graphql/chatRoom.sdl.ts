export const schema = gql`
  type ChatUser {
    id: Int!
    displayName: String!
  }

  type ChatMessage {
    id: Int!
    chatRoomId: String!
    user: ChatUser!
    body: String!
    createdAt: DateTime!
    # We could have properties for message for different user roles on the room, like isHidden
  }

  type Query {
    chatMessages(chatRoomId: String!): [ChatMessage!]! @skipAuth
  }

  input ChatMessageInput {
    body: String!
    chatRoomId: String!
  }

  type Mutation {
    sendChatMessage(input: ChatMessageInput!): ChatMessage! @requireAuth
  }
`
