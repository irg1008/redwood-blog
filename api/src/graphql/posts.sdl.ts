export const schema = gql`
  type Post {
    id: Int!
    title: String!
    slug: String!
    body: String!
    createdAt: DateTime!
    userId: Int!
    user: User!
  }

  type Query {
    posts: [Post!]! @skipAuth
    post(slug: String!): Post @skipAuth
  }
`
