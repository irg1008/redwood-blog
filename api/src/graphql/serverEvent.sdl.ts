export const schema = gql`
  input SendServerEventInput {
    userId: Int!
    message: String!
    topic: String!
  }

  type ServerEventResult {
    message: String!
    topic: String!
  }

  extend type Mutation {
    sendServerEvent(input: SendServerEventInput!): ServerEventResult!
      @requireWorker
  }
`
