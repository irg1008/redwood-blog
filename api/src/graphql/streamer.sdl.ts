export const schema = gql`
  type Streamer {
    id: Int!
    liveStreamId: Int
    liveStream: Stream
    streamPath: String!
    userId: Int!
    user: User!
  }

  type Query {
    streamer(id: Int!): Streamer @skipAuth
  }
`
