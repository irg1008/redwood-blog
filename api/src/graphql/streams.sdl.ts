export const schema = gql`
  type StreamKey {
    streamKey: String!
  }

  type StreamUrl {
    streamUrl: String!
  }

  type Query {
    readStream(streamId: Int!): StreamUrl! @skipAuth
  }

  input CreateStreamKeyInput {
    userId: Int!
  }

  type Mutation {
    adminCreateStreamKey(input: CreateStreamKeyInput!): StreamKey
      @requireAuth(roles: ["admin"])
    createStreamKey: StreamKey @requireAuth
  }
`
