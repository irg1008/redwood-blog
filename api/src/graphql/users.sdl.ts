export const schema = gql`
  type User {
    id: Int!
    name: String
    email: String!
    roles: Role!
    posts: [Post]!
  }

  enum Role {
    moderator
    admin
    user
  }

  type Query {
    users: [User!]! @requireAuth(roles: "admin")
    user(id: Int!): User @requireAuth
  }

  input ConfirmUserInput {
    email: String!
    code: Int!
  }

  type Mutation {
    confirmUser(input: ConfirmUserInput!): User @skipAuth
    sendConfirmCode(email: String!): Boolean @skipAuth
  }
`
