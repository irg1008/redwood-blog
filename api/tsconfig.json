{
  "compilerOptions": {
    "noEmit": true,
    "allowJs": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "target": "ES2023",
    "module": "Node16",
    "moduleResolution": "Node16",
    "skipLibCheck": false,
    "rootDirs": ["./src", "../.redwood/types/mirror/api/src"],
    "paths": {
      "src/*": ["./src/*", "../.redwood/types/mirror/api/src/*"],
      "types/*": ["./types/*", "../web/types/*"],
      "$web/*": ["../web/*"],
      "@redwoodjs/testing": ["../node_modules/@redwoodjs/testing/api"],
    },
    "typeRoots": ["../node_modules/@types", "./node_modules/@types"],
    "types": ["jest"],
    "jsx": "react-jsx"
  },
  "include": [
    "src",
    "../.redwood/types/includes/all-*",
    "../.redwood/types/includes/api-*",
    "../web/types",
    "./types"
  ]
}
