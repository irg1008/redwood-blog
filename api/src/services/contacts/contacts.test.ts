import { CreateContactInput } from 'types/graphql'

import { EmailValidationError } from '@redwoodjs/api'

import {
  contact,
  contacts,
  createContact,
  deleteContact,
  updateContact,
} from './contacts'
import type { StandardScenario } from './contacts.scenarios'

describe('contacts', () => {
  scenario('returns all contacts', async (scenario: StandardScenario) => {
    const result = await contacts()

    expect(result.length).toEqual(Object.keys(scenario.contact).length)
  })

  scenario('returns a single contact', async (scenario: StandardScenario) => {
    const result = await contact({ id: scenario.contact.john.id })

    expect(result).toEqual(scenario.contact.john)
  })

  scenario('creates a contact', async () => {
    const contact: CreateContactInput = {
      name: 'Jane Doe',
      email: 'jane@anonymous.com',
      message: 'Hi!',
    }

    const result = await createContact({ input: contact })

    expect(result.name).toEqual(contact.name)
    expect(result.email).toEqual(contact.email)
    expect(result.message).toEqual(contact.message)
    expect(result.createdAt).not.toEqual(null)
  })

  scenario('contact name is optional', async (scenario: StandardScenario) => {
    const result = await contact({ id: scenario.contact.nameless.id })
    expect(result.name).toBeNull()
  })

  scenario('updates a contact', async (scenario: StandardScenario) => {
    const original = await contact({ id: scenario.contact.john.id })

    const newName = 'Johnathan Doe'
    const result = await updateContact({
      id: original.id,
      input: { name: newName },
    })

    expect(result.name).toEqual(newName)
  })

  scenario('deletes a contact', async (scenario: StandardScenario) => {
    const original = await deleteContact({ id: scenario.contact.john.id })
    const result = await contact({ id: original.id })

    expect(result).toEqual(null)
  })

  scenario('only valid emails are inserted', async () => {
    expect(() =>
      createContact({
        input: {
          name: 'Jane Doe',
          email: 'jane@anonymous',
          message: 'Hi!',
        },
      })
    ).rejects.toThrow(EmailValidationError)
  })
})
