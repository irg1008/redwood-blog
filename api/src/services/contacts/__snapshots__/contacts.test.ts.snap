// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contacts creates a contact 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">

  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
    <meta name="x-apple-disable-message-reformatting" />
  </head>
  <div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">New contact form submitted<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div>
  </div>

  <body style="margin-left:auto;margin-right:auto;margin-top:auto;margin-bottom:auto;background-color:rgb(255,255,255);font-family:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;">
    <table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;margin-left:auto;margin-right:auto;margin-top:40px;margin-bottom:40px;border-radius:0.25rem;border-width:1px;border-style:solid;border-color:rgb(229,231,235);padding:20px">
      <tbody>
        <tr style="width:100%">
          <td>
            <h1 style="margin-left:0px;margin-right:0px;margin-top:30px;margin-bottom:30px;padding:0px;text-align:center;font-size:24px;font-weight:400;color:rgb(0,0,0)">New contact from Jane Doe &lt;jane@anonymous.com&gt;</h1>
            <hr style="width:100%;border:none;border-top:1px solid #eaeaea;margin-left:0px;margin-right:0px;margin-top:26px;margin-bottom:26px;border-width:1px;border-style:solid;border-color:rgb(234,234,234)" />
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">Hi!</p>
          </td>
        </tr>
      </tbody>
    </table>
  </body>

</html>"
`;

exports[`contacts creates a contact 3`] = `
"NEW CONTACT FROM JANE DOE <JANE@ANONYMOUS.COM>

--------------------------------------------------------------------------------

Hi!"
`;
